lib/posix/posix_internal.h

```diff
@@ -51,10 +52,9 @@ struct posix_thread {
 	int cancel_pending;
 	struct k_spinlock cancel_lock;
 
-	/* Pthread State */
-	enum pthread_state state;
-	pthread_mutex_t state_lock;
-	pthread_cond_t state_cond;
+	/* Pthread State (of type enum pthread_state) */
+	atomic_t state;
```

Comment:
Given that "most" accesses to this occur under the lock, would it be more efficient to require all of them to be under the lock and avoid using an atomic here? Atomic operations are quite expensive on some architectures, and not available on others (cortex M0, I'm looking at you).